### ISSUE TITLE
Row iterator returned by Table.where() query is incompatible with itertools.tee()

### ISSUE THREAD

**Issue by felker** [PyTables/PyTables#641]
I would like to take the Row iterator object returned by an in-kernel query, make an independent duplicate with `itertools.tee()`, consume one of the iterators, and then consume the second iterator. In other words,
```
row_iterator = table.query(condition)
row_iterator1, row_iterator2 = itertools.tee(row_iterator, 3)
for r in row_iterator1:
    print(r.nrow)
for r in row_iterator2:
   print(r.nrow)
```
The second loop returns nothing. I have read in other [issues](https://github.com/PyTables/PyTables/issues/267) here that the Row class `__iter__()` method is inherently stateful; does this explain the incompatibility? 

I know I could use `read` methods to cache features from the first row iterator rather than traverse an independent duplicate, and it would be faster and more efficient. However, I would like the flexibility to duplicate the row iterator for complex queries on small tables. It is unclear from the documentation that it would be incompatible with `itertools.tee()`

Python version = 2.7.13
PyTables version = 3.4.2

**Comment by FrancescAlted** [comment]
Yeah, I am afraid that this is an unfortunate consequence of the stateful iterators in PyTables.  Another good reason for refactoring them in a saner way.  As always, PR are very welcome.
